web/libs/ui/src/lib/sparkles/sparkles.tsx

```diff
@@ -0,0 +1,519 @@
+
+  // Helper to get a random position avoiding the center cutout
+  const getRandomPosition = React.useCallback(() => {
+    const center = { x: buttonSize / 2, y: buttonSize / 2 };
+    const options: SparkleAreaOptions = {
+      areaShape,
+      areaRadius: _areaRadius,
+      areaWidth: _areaWidth,
+      areaHeight: _areaHeight,
+      cutoutShape,
+      cutoutRadius: _cutoutRadius,
+      cutoutWidth: _cutoutWidth,
+      cutoutHeight: _cutoutHeight,
+      center,
+    };
+    return randomPositionAvoidingCenter(options);
+  }, [
+    areaShape,
+    _areaRadius,
+    _areaWidth,
+    _areaHeight,
+    cutoutShape,
+    _cutoutRadius,
+    _cutoutWidth,
+    _cutoutHeight,
+    buttonSize,
+  ]);
+
+  // Randomize interval for each sparkle cycle
+  const getRandomInterval = () => {
+    const base = Math.floor(
+      Math.random() * (sparkleBaseIntervalMax - sparkleBaseIntervalMin + 1) + sparkleBaseIntervalMin,
+    );
+    const jitter = Math.floor(Math.random() * (2 * sparkleJitter + 1) - sparkleJitter);
+    return Math.max(200, base + jitter);
+  };
+
+  const [interval, setIntervalState] = React.useState(getRandomInterval());
+
+  useRandomInterval(
+    () => {
+      const now = Date.now();
+      const nextSparkles = sparkles.filter((sp) => now - sp.createdAt < sparkleLifetime);
+      if (nextSparkles.length < sparkleCount) {
+        const size = Math.floor(Math.random() * (sparkleSizeMax - sparkleSizeMin + 1) + sparkleSizeMin);
+        const { top, left } = getRandomPosition();
+        const farEnough =
+          nextSparkles.length === 0 ||
+          nextSparkles.every((sp) => {
+            const dx = Number.parseFloat(sp.style.left) + sp.size / 2 - left;
+            const dy = Number.parseFloat(sp.style.top) + sp.size / 2 - top;
+            const dist = Math.sqrt(dx * dx + dy * dy);
+            const sizeDiff = Math.abs(sp.size - size);
+            // Relaxed: allow closer sparkles and smaller size differences
+            return dist >= Math.max(2, sparkleMinDistance / 2) && sizeDiff >= Math.max(1, sparkleMinSizeDiff / 2);
+          });
+        if (farEnough) {
+          nextSparkles.push({
+            id: `${Date.now()}-${Math.random()}`,
+            createdAt: Date.now(),
+            color,
+            size,
+            style: {
+              position: "absolute" as const,
+              top: `${top - size / 2}px`,
+              left: `${left - size / 2}px`,
+              pointerEvents: "none" as const,
+              zIndex: 2,
+            },
+          });
+        }
+      }
+      setSparkles(nextSparkles);
+      setIntervalState(getRandomInterval());
+    },
+    prefersReducedMotion ? null : interval,
+    prefersReducedMotion ? null : interval,
+  );
+
+  return (
+    <span
+      style={{
+        display: "inline-block",
+        position: "relative",
+        width: buttonSize,
+        height: buttonSize,
+        pointerEvents: "none",
+      }}
```

Comment:
Inline styles should be moved to tailwind and merged using `cnm` function utility with the className provided by the component props.